### ISSUE TITLE
Player Turns Invisible

### ISSUE THREAD

**Issue by TheWackersKian** [JimiIT92/UniverseGuard2#132]
Video: https://www.youtube.com/watch?v=aBku_wlBBc4

Description: When claiming a region there is a certain spot within that region (usually tends to be within the middle) where a player disappears when entering that chunk. When removing the region and relogging the chunk error is gone which is how I realised how it was this plugin. Within the video you can see my player at the left of the screen, sorry for the bad video haha I was split screening.

Here is the link to the region file too: https://pastebin.com/twdWPx7i

Thanks!

**Comment by TheIceBadger** [comment]
Had the same issue on my server, running latest version of the plugin.
It's a modded server for FTB Continuum with
```
Minecraft: 1.12.2
SpongeAPI: 7.1.0-4ab0d1a5
Sponge: 1.12.2-7.1.6-SNAPSHOT
Sponge Forge 1.12.2-2768-7.1.6-RC3684
Minecraft Forge: 14.23.5.2796
```

Claimed area would have entities go invisible around the center of it, seems to be the entire world height.
Disable the plugin and issue is gone.

**Comment by JimiIT92** [comment]
This is really really strange, as the plugin doesn't handle entity render code directly. Might be some side effect for something, will see if I can reproduce the issue and trace back to the cause

**Comment by TheWackersKian** [comment]
Yeah it is really strange. Is there an update on this case?

**Comment by JimiIT92** [comment]
Unfortunately not, I tried different approach in a local server but can't seem to replicate the case, even with the same region file :/

**Comment by TheWackersKian** [comment]
Basically the way I recreated this was to make a decently sized region, I don’t think it matters though. Then I expanded this region vertically. I don’t believe adding flags is necessary. It may be the expansion that does it?

**Comment by JimiIT92** [comment]
I don't know, but if it is it's really weird, since all the expansion does is modify the region points and nothing else... I'll see if doing these steps will reproduce the problem. What size was the region created?

**Comment by TheWackersKian** [comment]
Honestly I can’t remember the exact size but it was decently large. Although I have tested this bug multiple times and it seems it doesn’t rely on region size. It usually just occurs near the middle. But I recommend getting an account watching in the air and then an alt account just running around

**Comment by ivandis** [comment]
@TheWackersKian Are there any error messages in the console that are thrown when the player walks into that region?

**Comment by JimiIT92** [comment]
Just an update on this: I think I tried it all and got no luck reproducing the issue. I even tried with some friends to see if the issue was somehowe related to premium accounts, since all my tries with al accounts are done with third party launchers (thank you Mojang for not providing a proper Dev API). One I can think of is that it could be some setting related to entities render, in which case of course I have no control over it :/ Remember that MC stops rendering entities after some blocks or if they are like in the corner of the screen (for instance if you have a chest on your side and slowly turn your camera the other side you will see the chest disappears at some point). Let me know if you see any error or strange log in the console

**Comment by ivandis** [comment]
@TheWackersKian Another thing you could do is replicate the area, find the exact boundaries in which the entity becomes invisible, check if objects/terrain placed around the area are conditional factors on whether or not this occurs, and see if you can make this work with another claimed area.  Another thing worth checking is whether or not the zone you turn invisible is consistently the same size and position inside every claimed area you make.

**Comment by TheWackersKian** [comment]
Ok so I just downloaded the latest plugin update and now I can't seem to recreate the issue. I'm gonna test a few more things and if it doesn't happen then it's fixed. It was a really weird bug.

**Comment by JimiIT92** [comment]
If it does it kinda autofixed itsel 😂

**Comment by TheWackersKian** [comment]
Yeah all seems to be working. I will let you know if I run into it again

**Comment by TheWackersKian** [comment]
Ok yeah it's still happening. But once I disconnect and join back it fixes it self for a little bit

**Comment by TheWackersKian** [comment]
https://www.youtube.com/watch?v=TVRqd-aumEo This area is not even claimed with UniverseGuard either, but ever since I added it this player can no longer open his chests and we're having the invisible issue again

**Comment by ivandis** [comment]
Have you tried collecting the information I mentioned above?

**Comment by TheWackersKian** [comment]
Yes one thing I noticed quite peculiar is around chests. When I placed a chest it said something in chat about it being unprotected. Next thing an area next to it made the person become invisible. Quite weird

**Comment by ivandis** [comment]
Is this behavior variable depending on render distance?

**Comment by TheWackersKian** [comment]
I don't believe so as you can see in the video I'm within a few blocks of the player. Looking directly at him, that shouldn't really be render distance problems.

**Comment by ivandis** [comment]
I believe it would be worth checking.

**Comment by TheWackersKian** [comment]
How would you like me to test that? My render distance is already more than enough to render him. I'm confused haha

**Comment by JimiIT92** [comment]
I am concerned about that chest message you mentioned happens when placing a chest. Maybe it could be some conflict with other plugins/mods. Can you share a txt file with all the plugins/mods installed on the server (and their version)? Also please report if you made any changes to the respective configs, so I can set up an almost exact copy of your server and see if I can finally replicate the issue

**Comment by TheWackersKian** [comment]
Hmmmm maybe it could be GriefPrevention conflicting with UniverseGuard? I will work on that list now

**Comment by TheWackersKian** [comment]
https://www.dropbox.com/s/x5bwjxrichkree4/mods.zip?dl=0 This should make it easier. I uploaded all mods and plugins

**Comment by JimiIT92** [comment]
> it could be GriefPrevention conflicting with UniverseGuard?

Well, in general having two plugins for the same job is not ideal, so it could totally be. However I expect them to be in conflict for flag handlings, not rendering stuffs :/

**Comment by TheWackersKian** [comment]
Yeah flags work fine

**Comment by RocketManKian** [comment]
Any idea on this? I really need this plugin to work haha it's amazing. Everything I need

**Comment by JimiIT92** [comment]
I tried the modpack posted. If I try to setup a local server with that modpack I got this error
`Encountered an unexpected exception
net.minecraftforge.fml.common.MultipleModsErrored:
Missing Mods:
        unknown : need [14.23.5.2826,): have 14.23.5.2825`
        
While on client it says it need Pixelmon 8.0 or above, and in the modpack there is the 8.1.2 version... Is something missing from that folder?

**Comment by RocketManKian** [comment]
Let me send an updated link. My bad. https://www.dropbox.com/s/g8o3rxehto0mkj7/fortunecraft.zip?dl=0 these are the client mods.
